packages/components/src/internal/query/filter.ts

```diff
@@ -24,8 +24,9 @@ import {
     isRelativeDateFilterValue,
 } from '../util/Date';
 
-const QUERY_KEY_CHAR_DECODED = ['$', '/', '&', '}', '~', ',', '.'];
-const QUERY_KEY_CHAR_ENCODED = ['$D', '$S', '$A', '$B', '$T', '$C', '$P'];
+// N.B. the dollar sign should always be last here to avoid double-decoding (see getLegalIdentifier)
+const QUERY_KEY_CHAR_DECODED = ['/', '&', '}', '~', ',', '.', '$'];
```

Comment:
nit: Still seems odd to me we're repeating this here when we have `QueryKey.decodePart` and `QueryKey.encodePart`.

Reply:
I agree. I've removed this in favor of `QueryKey.decodePart`

Reply:
Great, thanks!